lib/Builtins.cpp

```diff
@@ -233,11 +233,14 @@ bool Builtins::FunctionInfo::GetFromMangledNameCheck(
   while (pos < mangled_name_len) {
     ParamTypeInfo type_info;
     if (mangled_name[pos] == 'S') {
-      // handle duplicate param_type
-      if (mangled_name[pos + 1] != '_') {
+      // handle duplicate param_type. Comes in two flavours:
+      // S_ and S#_.
+      char p1 = mangled_name[pos + 1];
+      char p2 = mangled_name[pos + 2];
```

Comment:
In isolation, I'm concerned about buffer overflow
Suggest guarding the access by pos+2 with the first test.


```
     if (p1 != '_' && (mangled_name[pos+2] != '_')) {
                return false;
     }
```